airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/JsonSchemaType.java

```diff
@@ -106,15 +110,29 @@ public static class Builder {
 
     private final ImmutableMap.Builder<String, Object> typeMapBuilder;
 
-    private Builder(final JsonSchemaPrimitive type) {
-      typeMapBuilder = ImmutableMap.builder();
-      if (JsonSchemaPrimitiveUtil.isV0Schema(type)) {
-        typeMapBuilder.put(TYPE, type.name().toLowerCase());
+    private Builder(final JsonSchemaPrimitive... types) {
+      final List<JsonSchemaPrimitive> schemaPrimitives = Arrays.asList(types);
+      if (hasMultipleTypes(schemaPrimitives)) {
```

Comment:
And there you can check for a `JsonSchemaType.JSON` and build the same oneof structure. 

When we update to V1, I can just make the change to have the V1 versions of `JsonSchemaType.STRING, JsonSchemaType.BOOLEAN, JsonSchemaType.NUMBER`